nixos/modules/services/libeufin/common.nix

```diff
@@ -58,54 +49,42 @@ libeufinComponent:
                 };
               in
               "${lib.getExe' cfg.package "libeufin-${libeufinComponent}"} serve ${args}";
+            Restart = "on-failure";
+            RestartSec = "10s";
           };
-          requires = [ "${serviceName}-dbinit.service" ];
-          after = [ "${serviceName}-dbinit.service" ];
+          requires = [ "libeufin-dbinit.service" ];
+          after = [ "libeufin-dbinit.service" ];
           wantedBy = [ "multi-user.target" ];
-
-          # TODO: Could this preStart block be part of the db-init service, and then
-          #       we add it as a dependency? It seems to be part of database initialization
-          #       in a certain sense
-
-          # Some accounts might need to be registered before the
-          # service starts, like the exchange when the bank's currency
-          # conversion is enabled.
-          preStart =
-            let
-              initialAccountRegistration = lib.concatMapStringsSep "\n" (
-                account:
-                let
-                  args = lib.cli.toGNUCommandLineShell { } {
-                    c = cfgMain.configFile;
-                    inherit (account) username password name;
-                    payto_uri = "payto://x-taler-bank/bank:${toString bankPort}/${account.username}?receiver-name=${account.name}";
-                    exchange = lib.toLower account.username == "exchange";
-                  };
-                in
-                "${lib.getExe' cfg.package "libeufin-bank"} create-account ${args}"
-              ) cfg.initialAccounts;
-            in
-            lib.mkIf ((!isNexus) && (cfg.initialAccounts != [ ])) ''
-              # only register initial accounts once
-              if [ ! -e /var/lib/${serviceName}/init ]; then
-                ${initialAccountRegistration}
-
-                touch /var/lib/${serviceName}/init
-                echo "Bank initialisation complete"
-              fi
-            '';
         };
 
         # Database Initialisation
-        "${serviceName}-dbinit" =
+        libeufin-dbinit =
```

Comment:
Will this now be defined twice, with different contents for both nexus and bank?

Reply:
The contents will be appended in one single service `libeufin-dbinit`. For example, the start script will initialize the bank, then the nexus:

```sh
#!/nix/store/717iy55ncqs0wmhdkwc5fg2vci5wbmq8-bash-5.2p32/bin/bash
set -e

/nix/store/4cvdvyaqpvcla1m7ks6n95ln33hncn4h-libeufin-0.13.0/bin/libeufin-bank dbinit -L debug -c /nix/store/lfhm2saxf9vjyrrdr5ni9p7pvg8abmym-generated-libeufin.conf

/nix/store/4cvdvyaqpvcla1m7ks6n95ln33hncn4h-libeufin-0.13.0/bin/libeufin-nexus dbinit -L debug -c /nix/store/lfhm2saxf9vjyrrdr5ni9p7pvg8abmym-generated-libeufin.conf
```

That's why we don't have to worry about them running at the same time and conflicting anymore.